lib/modules/submitHelper.js

```diff
@@ -51,41 +57,49 @@ const $repostWarning = _.once(() => $(string.html`
-			urlField = ((urlFieldDiv.querySelector('#url'): any): HTMLInputElement);
-			srField = ((document.querySelector('#sr-autocomplete'): any): HTMLInputElement);
-			$([srField, urlField]).on('input keydown', _.debounce(updateRepostWarning, 300));
-			// No event is fired when reddit's js changes the subreddit field, so update whenever the user clicks
-			$('#suggested-reddits a, #sr-drop-down').on('click', updateRepostWarning);
-			// We would allow reddit to show/hide the message for link/text posts with #link-desc
-			// but some subreddits hide this box, so we'll do it manually.
-			const linkButton = document.querySelector('a.link-button');
-			const textButton = document.querySelector('a.text-button');
-			if (linkButton && textButton) {
-				linkButton.addEventListener('click', () => { updateRepostWarning(); });
-				textButton.addEventListener('click', () => { $repostWarning().hide(); });
-			}
-		}
+	implementOptions();
+	// Register submit CLI if not on the submit page
+	if (!isPageType('submit')) {
+		registerCommandLine();
 	}
+};
 
-	if (isPageType('submit') && module.options.uncheckSendRepliesToInbox.value) {
-		const sendReplies: ?HTMLInputElement = (document.querySelector('#sendreplies'): any);
-		if (sendReplies) {
-			sendReplies.checked = false;
+function implementOptions() {
+	if (isPageType('submit')) {
+		// Implement submitHelper preferences if on the submit page
+		if (module.options.warnAlreadySubmitted.value) {
+			const urlFieldDiv = document.querySelector('#url-field');
+			if (urlFieldDiv) {
+				$(urlFieldDiv).parent().after($repostWarning());
+				urlField = ((urlFieldDiv.querySelector('#url'): any): HTMLInputElement);
+				srField = ((document.querySelector('#sr-autocomplete'): any): HTMLInputElement);
+				$([srField, urlField]).on('input keydown', _.debounce(updateRepostWarning, 300));
+				// No event is fired when reddit's js changes the subreddit field, so update whenever the user clicks
+				$('#suggested-reddits a, #sr-drop-down').on('click', updateRepostWarning);
+				// We would allow reddit to show/hide the message for link/text posts with #link-desc
+				// but some subreddits hide this box, so we'll do it manually.
+				const linkButton = document.querySelector('a.link-button');
+				const textButton = document.querySelector('a.text-button');
+				if (linkButton && textButton) {
+					linkButton.addEventListener('click', () => { updateRepostWarning(); });
+					textButton.addEventListener('click', () => { $repostWarning().hide(); });
+				}
+			}
 		}
-	}
 
-	if (isPageType('submit') && module.options.focusFormOnLoad.value) {
-		$('form.submit [name=url], form.submit [name=title]').filter(':visible').first().focus();
-	}
+		if (module.options.uncheckSendRepliesToInbox.value) {
+			const selector = isAppType('d2x') ? '[aria-labelledby="send-replies"]' : '#sendreplies';
+			const sendReplies: ?HTMLElement = document.querySelector(selector);
+			if (sendReplies) {
```

Comment:
As you mentioned, this will probably be null immediately after `urlChanged` jsapi event fires since Reddit's react component hasn't mounted yet. In this case, we could adopt the selenium approach of "await element".

If RES already had a helper function, I would look in the `dom` utils. If not, here's an idea:

```javascript
function awaitElement(selector, container = document.documentElement, timeout = 5000) {
    const rejectAfter = Date.now() + timeout;
    return new Promise ((resolve, reject => {
        f();
        function f() {
              const element = container.querySelector(selector);
                if (element) {
                     resolve(element);
                 } else if (Date.now() >= rejectAfter) {
                      reject("timeout");
                 } else {
                      setTimeout(f, 100);
                 }
            }
    });
}
```